lib/excalt/vcard/req_builder.ex

```diff
@@ -68,7 +68,60 @@ defmodule Excalt.Vcard.ReqBuilder do
     |> Saxy.encode!()
   end
 
-  #  def get_contacts(multiple_urls) do
+  # Expect multiple urls in a form of list of strings
+  # e.g. ["url1", "url2", "url3"]
+  def get_contacts(multiple_urls) do
+    xhref_elements = build_multiple_xml_href_elements(multiple_urls)
 
-  #  end
+    xml_elements =
+      [
+        Saxy.XML.element("D:prop", [], [
+          Saxy.XML.element("card:address-data", [], ""),
+          Saxy.XML.element("D:getetag", [], "")
+        ])
+      ]
+      |> append_urls_to_elements(xhref_elements)
+
+    Saxy.XML.element(
+      "card:addressbook-multiget",
+      [
+        "xmlns:D": "DAV:",
+        "xmlns:card": "urn:ietf:params:xml:ns:carddav"
+      ],
+      xml_elements
+    )
+    |> Saxy.encode!()
+  end
+
+  @doc """
+  Getting only etags from the server and comparing to one we
+  already have can save us a ton of bandwith. The server should return the
+  response off all etags and urls from the addressbook, so we can compare which
+  of them are created/updated/deleted by other clients, and we can update our
+  local addressbook with newest changes.
+  """
+  def get_etags() do
```

Comment:
please add another function to fetch the etag of on individual vcard.

Reply:
After a thinking for a while, there is no need to make another xml request to fetch individual contact with etag. The simple get request is enough for that purpose.